screens/Tractors/AddEditTractor.js

```diff
@@ -10,48 +13,116 @@ export const MyButton = ({value="Add/Edit",cb={}}) => {
     )
 }
 
-const AddEditTractor = () => {
+const AddEditTractor = ({route}) => {
+    const navigation = useNavigation()
+    const [tractorState,dispatch] = useTractor()
+   
+    const [id,setId] = useState("")
+    const [Name,setName] = useState("")
+    const [Size,setSize] = useState("")
+    const [model,setModel] = useState("")
+    const [company,setCompany] = useState("")
+    const [registrationNumber,setRegistrationNumber] = useState("")
+    const [color,setColor] = useState("")
+   var type = route.params?.type
+
+   useEffect(() => {
+    const current = tractorState.current
+    if(type=="edit"){
+                setId(current.id)
+                setName(current.name)
+                setSize(current.size)
+                setModel(current.model)
+                setCompany(current.company)
+                setRegistrationNumber(current.registration)
+                setColor(current.color)
+           }
+           else{
+            setId(uuid.v4())
+           }
+   },[])
+   const OnSubmit = () => {
+    if(Name.trim() === "" || company.trim()=== "" || registrationNumber.trim()=== ""){
+        ToastAndroid.show("Please fill required fields",ToastAndroid.SHORT)
```

Comment:
So if we want to handle for iOS, do we need to add check here?

Reply:
Yes, for iOS we use AlertIOS hook for the toast message here.